core/worker.go

```diff
@@ -976,6 +978,31 @@ func (w *worker) commit(env *environment, interval func(), update bool, start ti
 	return nil
 }
 
+// pendingBodyKey comprises all root fields of the header
+type pendingBodyKey struct {
+	UncleHash common.Hash
+	Root      common.Hash
```

Comment:
I don't think `Root` should be here. It over constrains and may yet cause other trip ups.

`TxHash`, `UncleHash`, and `EtxHash` perfectly uniquely describe the body (except for the submanifest) as desired.

Reply:
Good catch. I will remove it as it is extra computation. I think root is not needed too. 

But I do believe it doesn’t cause any problem since other three keys make it unique so adding a constant or a variable doesn’t do anything to the uniqueness of the key.

Reply:
changed it